src/Servers/Kestrel/Core/src/Internal/CertificatePathWatcherLoggerExtensions.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
+
+internal static partial class CertificatePathWatcherLoggerExtensions
+{
+    [LoggerMessage(0, LogLevel.Warning, @"Directory ""{dir}"" does not exist so changes to the certificate ""{path}"" will not be tracked", EventName = "DirectoryDoesNotExist")]
```

Comment:
Nit: I don't think we usually 0-index event IDs.

Reply:
Structured log names are also capitalized and usually fully named (and maybe descriptive). `{dir}` -> `{CertificateDirectory}`, `{path}` -> `{CertificatePath}`